pool/app/experiment2/model.ml

```diff
@@ -0,0 +1,187 @@
+open Pool_common
+
+type experiment_type = Pool_common.ExperimentType.t =
+  | Lab
+  | Online
+[@@deriving eq, show]
+
+type t =
+  { id : Id.t
+  ; title : string
+  ; public_title : string
+  ; description : string option
+  ; cost_center : string option
+  ; organisational_unit : Organisational_unit.t option
+  ; filter : Filter.t option
+  ; contact_person_id : Id.t option
+  ; smtp_auth_id : Id.t option
+  ; direct_registration_disabled : bool
+  ; registration_disabled : bool
+  ; allow_uninvited_signup : bool
+  ; external_data_required : bool
+  ; show_external_data_id_links : bool
+  ; experiment_type : experiment_type
+  ; email_session_reminder_lead_time : Non_negative_time_span.t option
+  ; text_message_session_reminder_lead_time : Non_negative_time_span.t option
+  ; invitation_reset_at : Time_span.t option
+  ; created_at : Ptime.t
+  ; updated_at : Ptime.t
+  }
+[@@deriving eq, show]
+
+let id t = t.id
+let title t = t.title
+let public_title t = t.public_title
+let description t = t.description
+let cost_center t = t.cost_center
+let organisational_unit t = t.organisational_unit
+let filter t = t.filter
+let contact_person_id t = t.contact_person_id
+let smtp_auth_id t = t.smtp_auth_id
+let direct_registration_disabled t = t.direct_registration_disabled
+let registration_disabled t = t.registration_disabled
+let allow_uninvited_signup t = t.allow_uninvited_signup
+let external_data_required t = t.external_data_required
+let show_external_data_id_links t = t.show_external_data_id_links
+let experiment_type t = t.experiment_type
+let email_session_reminder_lead_time t = t.email_session_reminder_lead_time
+let invitation_reset_at t = t.invitation_reset_at
+let created_at t = t.created_at
+let updated_at t = t.updated_at
+
+let text_message_session_reminder_lead_time t =
+  t.text_message_session_reminder_lead_time
+;;
+
+let public_title_placeholder = "###"
+
+let sql_select_fragment =
+  [%string
```

Comment:
Is this needed to evaluate the sql_select_fragments?

Reply:
Yup, a few lines below we have:

```ocaml
     {sql|
        %{ Id.sql_select_fragment ~field:"pool_experiments.contact_person_uuid" },
        %{ Id.sql_select_fragment ~field:"pool_experiments.smtp_auth_uuid" },
        %{ Id.sql_select_fragment ~field:"pool_experiments.uuid" },
        ...
     |sql}
```

where are using this ppx to compose these strings without needing to always bring `Format.sprintf`